OpenRA.Mods.Common/Widgets/ObserverArmyIconsWidget .cs

```diff
@@ -0,0 +1,220 @@
+
+		int lastIconIdx;
+
+		[ObjectCreator.UseCtor]
+		public ObserverArmyIconsWidget(World world, WorldRenderer worldRenderer)
+		{
+			this.world = world;
+			this.worldRenderer = worldRenderer;
+
+			GetTooltipUnit = () => TooltipUnit;
+			tooltipContainer = Exts.Lazy(() =>
+				Ui.Root.Get<TooltipContainerWidget>(TooltipContainer));
+
+			Init();
+		}
+
+		protected ObserverArmyIconsWidget(ObserverArmyIconsWidget other)
+			: base(other)
+		{
+			GetPlayer = other.GetPlayer;
+			world = other.world;
+			worldRenderer = other.worldRenderer;
+
+			IconWidth = other.IconWidth;
+			IconHeight = other.IconHeight;
+			IconSpacing = other.IconSpacing;
+			iconSize = new float2(IconWidth, IconHeight);
+
+			MinWidth = other.MinWidth;
+
+			TooltipUnit = other.TooltipUnit;
+			GetTooltipUnit = () => TooltipUnit;
+
+			TooltipTemplate = other.TooltipTemplate;
+			TooltipContainer = other.TooltipContainer;
+
+			tooltipContainer = Exts.Lazy(() =>
+				Ui.Root.Get<TooltipContainerWidget>(TooltipContainer));
+
+			Init();
+		}
+
+		void Init()
+		{
+			var buildable = world.Map.Rules.Actors.Values
+				.Where(x =>
+					x.Name[0] != '^' &&
+					x.HasTraitInfo<BuildableInfo>())
+				.SelectMany(x => x.TraitInfo<BuildableInfo>().Queue, (ai, q) => new Tuple<string, ActorInfo, BuildableInfo>(q, ai, ai.TraitInfo<BuildableInfo>()));
+
+			var pqueue = world.Map.Rules.Actors.Values
+				.SelectMany(a => a.TraitInfos<ProductionQueueInfo>());
+
+			unitLookup = buildable.Join(pqueue, tuple => tuple.Item1, pq => pq.Type, (tuple, info) => new { tuple.Item2.Name, Ai = tuple.Item2, tuple.Item3.BuildPaletteOrder, info.DisplayOrder })
+				.Distinct()
+				.ToDictionary(t => t.Name, o => new Tuple<ActorInfo, int, int>(o.Ai, o.DisplayOrder, o.BuildPaletteOrder));
+		}
+
+		public override void Draw()
+		{
+			armyUnits.Clear();
+			armyIconsBounds.Clear();
+
+			var player = GetPlayer();
+			if (player == null)
+				return;
+
+			var playerStatistics = player.PlayerActor.TraitOrDefault<PlayerStatistics>();
+
+			var items = playerStatistics.Units
+				.Where(i => i.Value.Count > 0)
+				.OrderBy(i => unitLookup[i.Key].Item2)
+				.ThenBy(i => unitLookup[i.Key].Item3)
+				.ToList();
+
+			Bounds.Width = items.Count * (IconWidth + IconSpacing);
+
+			var queueCol = 0;
+			foreach (var unit in items)
+			{
+				var name = unit.Key;
+				var actorInfo = unitLookup[name].Item1;
+
+				var rsi = actorInfo.TraitInfo<RenderSpritesInfo>();
+				var icon = new Animation(world, rsi.GetImage(actorInfo, world.Map.Rules.Sequences, player.Faction.Name));
+				var bi = actorInfo.TraitInfo<BuildableInfo>();
+				icon.Play(bi.Icon);
+				var topLeftOffset = new float2(queueCol * (IconWidth + IconSpacing), 0);
+
+				var iconTopLeft = RenderOrigin + topLeftOffset;
+				var centerPosition = iconTopLeft;
+
+				WidgetUtils.DrawSHPCentered(icon.Image, centerPosition + 0.5f * iconSize, worldRenderer.Palette(bi.IconPalette), 0.5f);
```

Comment:
I don't understand why, but this seems to be using the wrong palette?
Notice how the pure black in the mobile flak icon is drawn as transparent:
![Screenshot 2019-07-20 at 11 09 31](https://user-images.githubusercontent.com/167819/61577438-18ae0f00-aadf-11e9-8f43-b405acf6f30a.png)

Reply:
The Mobile Flak cameo does use color #0 on those spots. It's an asset error overlooked until now, I guess.